### ISSUE TITLE
Indeterminate Checkbox on Safari w/ VO on MacOS reports state incorrectly

### ISSUE THREAD

**Issue by jnurthen** [adobe/react-spectrum#4004]
# 🐛 Bug Report 

An isIndeterminate checkbox reports "unchecked" when in the mixed state. If you check using the keyboard it then reports checked (and still looks indeterminate) and then check it again and it reports unchecked (still looks indeterminate)

## 🤔 Expected Behavior

The state should be the same as that shown visually

## 😯 Current Behavior

https://reactspectrum.blob.core.windows.net/reactspectrum/41e26fa0f01ee9ab6e5c21edf6f7bb5d498bf40f/storybook/index.html?path=/story/checkbox--isindeterminate-true&providerSwitcher-express=false&providerSwitcher-toastPosition=bottom
* Using Safari, Start VO
* move to the checkbox and note it is read incorrectly

## 💁 Possible Solution

I think this is a safari bug as I can reproduce on the indeterminate example at https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/checkbox#examples

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   |
| Browser          | Safari 16.3
| Operating System | MacOS 13.2

**Comment by devongovett** [comment]
Does `aria-checked` fix it? We had it until #3130. We could revert #3687 if that fixes it.

**Comment by jnurthen** [comment]
apparently not according to https://github.com/vaadin/web-components/issues/2929

**Comment by jnurthen** [comment]
Interestingly this seems to be a webkit issue as this works ok with Chrome+VO

**Comment by twilco** [comment]
This should be fixed with Safari Technology Preview 161:

https://developer.apple.com/safari/technology-preview/release-notes/

> Fixed checkboxes not respecting indeterminate state ([258314@main](https://commits.webkit.org/258314@main))

If you try STP and still experience this bug, please do post back here (or on [bugs.webkit.org](https://bugs.webkit.org)).

**Comment by jnurthen** [comment]
Thanks - verified.

**Comment by LFDanLu** [comment]
@jnurthen if `aria-checked` doesn't fix this, do you know of any other ways we could handle this for older Safari versions?

**Comment by snowystinger** [comment]
Closing as there's nothing to be done for older versions of Safari and newer ones work correctly.
